runtime/vm/mz64/UpcallThunkGen.cpp

```diff
@@ -328,33 +329,28 @@ createUpcallThunk(J9UpcallMetaData *metaData)
 				if (fprIndex <= maxFPRIndex - 2) {
 					thunkMem += STEY(thunkMem, fprIndex, 0, R4, argAreaOffsetFromSP + (currStackSlotIndex * 8));
 					thunkMem += STEY(thunkMem, fprIndex + 2, 0, R4, argAreaOffsetFromSP + (currStackSlotIndex * 8) + 4);
-					currStackSlotIndex += 1;
 					fprIndex += 4;
 					if (i < 3) {
 						gprIndex += 2;
 					}
 				}
 				else if (fprIndex <= maxFPRIndex) {
 					thunkMem += STEY(thunkMem, fprIndex, 0, R4, argAreaOffsetFromSP + (currStackSlotIndex * 8));
-					currStackSlotIndex += 1;
 					fprIndex += 2;
 				}
 			}
 			else { // All other pure float structs are passed in GPR (if there's space)
 				if (gprIndex <= maxGPRIndex) {
-					int totalStackSlotsReq = ((argSizeInBytes - 1) / 8) + 1;
-					int numGPRsAvailable = maxGPRIndex - gprIndex + 1;
-					int tempArgIndex = currStackSlotIndex;
+					U_32 numGPRsAvailable = maxGPRIndex - gprIndex + 1;
+					U_32 tempArgIndex = currStackSlotIndex;
 						for (int i = 1; i <= std::min(numGPRsAvailable, totalStackSlotsReq); i++) {
```

Comment:
This loop should use `U_32` to avoid a signed vs. unsigned comparison. Same comment for lines 378 and 401.

Reply:
I've addressed this comment and the `switch{` comment in: https://github.com/eclipse-openj9/openj9/pull/20322/commits/6fdda30d889262d4ef95afc8c8ccfd4cd5bea3e6. Once you've approved the PR I'll squash the commits.

Reply:
Thank you. Please squash.

Reply:
Done. Thanks!